src/sage/rings/polynomial/multi_polynomial_sequence.py

```diff
@@ -754,33 +828,18 @@ def coefficient_matrix(self, sparse=True):
             [  b]
             [  c]
             [  d]
-            [  1]
             sage: A*v
             [        a + 2*b + 2*c + 2*d - 1]
             [a^2 + 2*b^2 + 2*c^2 + 2*d^2 - a]
             [      2*a*b + 2*b*c + 2*c*d - b]
             [        b^2 + 2*a*c + 2*b*d - c]
         """
-        R = self.ring()
-
-        m = sorted(self.monomials(),reverse=True)
-        nm = len(m)
-        f = tuple(self)
-        nf = len(f)
-
-        #construct dictionary for fast lookups
-        v = dict( zip( m , range(len(m)) ) )
-
         from sage.matrix.constructor import Matrix
+        from sage.misc.superseded import deprecation
 
-        A = Matrix( R.base_ring(), nf, nm, sparse=sparse )
-
-        for x in range( nf ):
-            poly = f[x]
-            for y in poly.monomials():
-                A[ x , v[y] ] = poly.monomial_coefficient(y)
-
-        return A, Matrix(R,nm,1,m)
+        deprecation(37035,"the function 'coefficient_matrix' is deprecated; use 'coefficients_monomials' instead")
+        cm = self.coefficients_monomials()
```

Comment:
It is really minor, but I think `A, v = self.coefficients_monomials()` would be clearer.